gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/recovery/RecoveryHelper.java

```diff
@@ -91,15 +91,13 @@ public boolean persistFile(State state, CopyableFile file, Path path) throws IOE
     }
 
     String guid = computeGuid(state, file);
-    StringBuilder nameBuilder = new StringBuilder(guid);
-    nameBuilder.append("_");
-    nameBuilder.append(shortenPathName(file.getOrigin().getPath(), 250 - nameBuilder.length()));
 
     if (!this.fs.exists(this.persistDir.get())) {
       this.fs.mkdirs(this.persistDir.get(), new FsPermission(FsAction.ALL, FsAction.READ, FsAction.NONE));
     }
 
-    Path targetPath = new Path(this.persistDir.get(), nameBuilder.toString());
+    Path targetPath = new Path(new Path(this.persistDir.get(), guid),
```

Comment:
I believe you first have to ensure the parent directory exists or the rename will fail (in some FileSystems).

Reply:
Changed mkdirs to create parent dir if it doesn't exist